src/tox/venv.py

```diff
@@ -226,20 +239,32 @@ def matching_platform(self):
         return re.match(self.envconfig.platform, sys.platform)
 
     def finish(self):
-        self._getliveconfig().writeconfig(self.path_config)
+        previous_config = CreationConfig.readconfig(self.path_config)
+        live_config = self._getliveconfig()
+        if previous_config is None or not previous_config.matches(live_config):
+            self.session.report.verbosity1(
+                "write config to {} as {!r}".format(self.path_config, live_config)
+            )
+            live_config.writeconfig(self.path_config)
 
     def _needs_reinstall(self, setupdir, action):
         setup_py = setupdir.join("setup.py")
         setup_cfg = setupdir.join("setup.cfg")
         args = [self.envconfig.envpython, str(setup_py), "--name"]
         env = self._get_os_environ()
         output = action.popen(args, cwd=setupdir, redirect=False, returnout=True, env=env)
-        name = output.strip()
-        args = [self.envconfig.envpython, "-c", "import sys; print(sys.path)"]
+        name = next(
+            (i for i in output.split("\n") if i and not i.startswith("pydev debugger:")), ""
```

Comment:
is this left over from a debugging session?

Reply:
More like a small fixup 👍 Ideally people should be able to run the test suite in debug mode too with pydev (both eclipse and Pycharm). For that to work this is needed.